docs/source/rag/retrieval_question_generation.rst

```diff
@@ -0,0 +1,208 @@
+.. _retrieval-question-generation:
```

Comment:
Can we structure this PR as a tutorial on top of the [docs-refactor](https://github.com/mlflow/mlflow/tree/docs-refactor) branch? @prithvikannan and @BenWilson2 can help (Ben is OOO but will be back Monday)

Reply:
I think this RAG should be placed within the LLM guide. See https://output.circle-artifacts.com/output/job/22eaa5b9-c5c4-41ff-b571-a3727fd89bc9/artifacts/0/docs/build/html/llms/index.html

Reply:
Can this PR be rebased and set to merge into the docs-refactor branch? That way you can use the notebook navigation CSS (to make things consistent and obvious about where to go to see the notebook & download the notebook locally)?